AsyncDisplayKit/ASCellNode+Internal.h

```diff
@@ -58,10 +60,7 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, strong, nullable) UICollectionViewLayoutAttributes *layoutAttributes;
 
-/// readwrite variant of the readonly public property.
-@property (nonatomic, copy, nullable) NSString *supplementaryElementKind;
-
-@property (nonatomic, copy, nullable) NSIndexPath *cachedIndexPath;
+@property (weak, nullable) ASCollectionElement *collectionElement;
```

Comment:
True – right now the Element owns the Node. That way node stays alive as long as someone has a reference to its collection element, but if you take a node and put it somewhere else, then we let the element die if nobody else retains it.